check language file

Deardrops opened the pull request:
测试：请随意修改除`en.hjson`以外的lang文件中的key，然后 `npm run test:lang`

Deardrops committed:
check language file

Deardrops committed:
var -> const

now jslint passed

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,50 @@
+
+const rf = require('fs')
+
+const lang_dir = './src/lang/'
```
变量名不要下划线

webxoss replied:
为什么要加`lang`前缀？`dir`能造成任何误解吗？

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const rf = require('fs')
```
rf什么鬼

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,50 @@
+  'ko',
+  'ru',
+  'zh_Hans',
+]
```
写死？不去读目录吗？

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,50 @@
+
+const _ = require('lodash')
+
+function check_keys(name) {
```
命名风格谁教的，一会儿大写开头一会儿驼峰一会儿下划线

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,50 @@
+  const keys = Object.keys(lang_maps[name])
+  const origin_keys = Object.keys(lang_maps['en'])
+  const surplus_keys = _.difference(keys, origin_keys)
+  if (surplus_keys.length !== 0) {
```
`if (length)`
不要和0比，我记得我之前的reivew说过的

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,50 @@
+  check_keys(name)
+})
+
+console.log('All Done!')
```
错误也正常退出？
`process.exit(code)`呢？

webxoss commented on package.json:
```diff
@@ -9,7 +9,8 @@
     "lint:node": "eslint . --c eslint.config.js --ignore-path .gitignore --cache --ignore-pattern src",
     "serve": "npm run lint:node && node serve",
-    "test": "npm run lint && node test"
+    "test": "npm run lint && node test",
```
把 build test 和 lang test 整合起来

Deardrops committed:
rename varables

remove varbales' prefix
exit(1) when check language fail
merge check language into test

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+const dir = './src/lang/'
+
+let names = []
+
```
建议这里不要空行

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+  names = fs.readdirSync(dir)
+}
+
+const Hjson = require('hjson')
```
建议移到顶部

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+const Hjson = require('hjson')
+
+let maps = {}
+
```
建议这里不空行

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+
+const dir = './src/lang/'
+
+let names = []
```
用了let就是不打算兼容旧版本node吧？那为什么不用箭头函数，字符模板？

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+  maps[name] = json
+})
+
+const _ = require('lodash')
```
库建议移到顶部，只有一两个地方用到的建议不设全局变量或干脆不要变量

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+
+const _ = require('lodash')
+
+let checkFail = false
```
加ed，你英语考多少分

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,58 @@
+if (checkFail) {
+  process.exit(1)
+} else {
+  console.log('All Done!')
```
无意义的Log,
log点有用的信息吧，至少把测试了什么说清楚啊

Deardrops committed:
es6 code style

use string template, =>
rename checkFail -> checkFailed
move require to top

webxoss commented on check_language.js:
```diff
@@ -0,0 +1,53 @@
+  let originKeys = Object.keys(maps['en.hjson'])
+  let surplusKeys = _.difference(keys, originKeys)
+  if (surplusKeys.length) {
+    console.warn(`Surplus keys in ${name}`)
```
这里要用error吧

webxoss commented on package.json:
```diff
@@ -9,7 +9,8 @@
     "lint:node": "eslint . --c eslint.config.js --ignore-path .gitignore --cache --ignore-pattern src",
     "serve": "npm run lint:node && node serve",
-    "test": "npm run lint && node test"
+    "test": "npm run lint && node test && npm run check:language",
```
把测试脚本（包括构建测试）都扔到 test/*.js
写个总的 test.js 遍历目录下的测试脚本跑

或者来个配置文件里面写要跑的测试脚本

webxoss replied:
都扔一个目录遍历的话就得依赖文件名来指定顺序了，一般先跑耗时短的脚本比较好

写配置的话增减改测试脚本的话又要跑去改，关注点分离了

嘛，你看怎么弄吧

Deardrops committed:
move test file into ./test folder

Deardrops commented on test/index.js:
```diff
@@ -0,0 +1,3 @@
+'use strict'
+require('./language.js')
+require('./complie.js')
```
测试文件很少增删，出于这样的考虑，选择在这里一一列出

Deardrops committed:
Add test commands

webxoss commented:
`npm run test language`
这样，不是多个 script ，是这个 script 接受参数

Deardrops committed:
specify test by 'npm run test xxx'

webxoss commented on eslint.config.js:
```diff
@@ -2,7 +2,11 @@
+let env = process.env.npm_lifecycle_event
+if (env) {
+  env = env.replace('lint:', '')
+}
```
？？？

webxoss commented on test/index.js:
```diff
@@ -0,0 +1,20 @@
+    require(tests[name])
+  }
+} else {
+  let argv = args[0]
```
应该是`arg`吧？
话说`v`是什么意思呢？

webxoss commented on test/language.js:
```diff
@@ -0,0 +1,53 @@
+  process.exit(1)
+} else {
+  console.log(names)
+  console.log('All language file test passed!')
```
加s，你是怎么过6级的

webxoss commented on test/language.js:
```diff
@@ -0,0 +1,53 @@
+  let originKeys = Object.keys(maps['en.hjson'])
+  let surplusKeys = _.difference(keys, originKeys)
+  if (surplusKeys.length) {
+    console.error(`Surplus keys in ${name}`)
```
感觉应该叫`unexpected keys`

webxoss commented on test/language.js:
```diff
@@ -0,0 +1,53 @@
+  }
+  let lackKeys = _.difference(originKeys, keys)
+  if (lackKeys.length) {
+    console.error(`Lack keys in ${name}`)
```
感觉应该叫`missing keys`

Deardrops committed:
env === 'node' when npm run lint

rename varables

webxoss commented on test/language.js:
```diff
@@ -24,16 +24,16 @@ let checkFailed = false
-  if (surplusKeys.length) {
+  let unexpectKeys = _.difference(keys, originKeys)
+  if (unexpectKeys.length) {
     console.error(`Surplus keys in ${name}`)
```
你 log 不改的嘛。。
